src/library/dao/playlistdao.cpp

```diff
@@ -246,6 +246,29 @@ bool PlaylistDAO::isPlaylistLocked(const int playlistId) const {
     return false;
 }
 
+bool PlaylistDAO::clearPlaylist(const int playlistId) {
+    int position =
+        (m_pAutoDJProcessor && m_pAutoDJProcessor->nextTrackLoaded()) ? 2 : 1;
```

Comment:
Yes, AutoDJProcessor relies on the currently playing track being at the top of the ADJ queue.